stdlib/std.jsonnet

```diff
@@ -768,11 +768,24 @@ limitations under the License.
               error 'Cannot use * precision with object.'
             else
               code.prec;
-          local val =
-            if std.objectHasAll(obj, f) then
-              obj[f]
+          local get_nested_val(obj, f, start=0, end=0) =
+            if !std.isObject(obj) then
+              error 'Subfield ' + f[:start] + ' didn\'t yield an object.'
+            else if end >= std.length(f) then
+              if std.objectHasAll(obj, f[start:]) then
+                obj[f[start:]]
+              else
+                error 'No such field: ' + f
+            else if f[end] == "." then
+              local key = f[start:end];
+              if std.objectHasAll(obj, key) then
+                get_nested_val(obj[key], f, end+1, end+1) tailstrict
+              else
+                error 'No such partial field: ' + f[:end]
             else
-              error 'No such field: ' + f;
+              get_nested_val(obj, f, start, end+1) tailstrict
+          ;
+          local val = get_nested_val(obj, f);
```

Comment:
Oh, this thing breaks some of my code, where I use formatting on object with '.' in field names, i.e
`"%(a.b)d" % {'a.b': 2}`
Shouldn't this be
`local val = if std.objectHasAll(obj, f) then obj[f] else get_nested_val(obj, f)`
?

Reply:
That's probably better given jsonnet is allowing '.' in field names already but is a little hacky. Probably better than current code though to keep backwards compatibility. I was thinking of mimicking this after python's f'{a.b}' which wouldn't be interpreted as a single field so I just never considered that.

@sparkprime should probably decide which approach is more important but I think supporting the existing use case makes sense.

Reply:
fortunate that you did not have a `)` in your field name I guess :)

I suppose the better fix is to have a way to put arbitary expressions inside strings, like you can do with javascript now.

Reply:
I've made an attempt at looking up values with JSONPath, it's not exactly the same as it moves the expression out of the string but should accomplish the same results. 

Have a look at it here: https://github.com/jsonnet-libs/xtd/blob/master/docs/jsonpath.md#fn-getjsonpath